web/src/components/StatusLoader.tsx

```diff
@@ -0,0 +1,193 @@
+  0.32307 * Dimension * 2
+}`
+const CheckmarkOffset = 0.92307 * Dimension * 2
+
+export type StatusLoaderState = 'loading' | 'success' | 'error'
+
+export function StatusLoader({
+  state,
+  message,
+}: {
+  state: StatusLoaderState
+  message?: string
+}) {
+  const rotateControls = useAnimation()
+  const circleControls = useAnimation()
+  const checkmarkControls = useAnimation()
+  const x1Controls = useAnimation()
+  const x2Controls = useAnimation()
+  const textControls = useAnimation()
+
+  const prevState = usePrevious<StatusLoaderState>(state)
+
+  React.useEffect(() => {
+    if (state !== prevState) {
+      if (state === 'loading') {
+        circleControls.start({
+          stroke: Theme.primary.main,
+          strokeDashoffset: Circumference * 0.75,
+        })
+        rotateControls.stop()
+        rotateControls.start({
+          rotate: [0, 360],
+        })
+
+        if (prevState === 'success') {
+          checkmarkControls.start({
+            strokeDashoffset: CheckmarkOffset,
+          })
+          textControls.start({
+            opacity: 0,
+          })
+        }
+
+        if (prevState === 'error') {
+          x1Controls.start({
+            strokeDashoffset: 2 * XOffset,
+          })
+          x2Controls.start({
+            strokeDashoffset: 2 * XOffset,
+          })
+          textControls.start({
+            opacity: 0,
+          })
+        }
+      }
+
+      if (prevState === 'loading' && state === 'success') {
+        // Stop rotate animation
+        rotateControls.stop()
+        // Transition in checkmark
+        checkmarkControls.start({
+          stroke: Theme.semantics.success,
+          strokeDashoffset: 0,
+        })
+        // Transition in full circle
+        circleControls.start({
+          stroke: Theme.semantics.success,
+          strokeDashoffset: 0,
+        })
+        textControls.start({
+          opacity: 1,
+          color: Theme.semantics.success,
+        })
+      }
+
+      if (prevState === 'loading' && state === 'error') {
+        // Stop rotate animation
+        rotateControls.stop()
+        // Transition in cross
+        x1Controls.start({
+          stroke: Theme.semantics.error,
+          strokeDashoffset: 0,
+        })
+        x2Controls.start({
+          stroke: Theme.semantics.error,
+          strokeDashoffset: 0,
+        })
+
+        // Transition in full circle
+        circleControls.start({
+          stroke: Theme.semantics.error,
+          strokeDashoffset: 0,
+        })
+
+        textControls.start({
+          opacity: 1,
+          color: Theme.semantics.error,
+        })
+      }
+    }
+
+    return () => {
+      rotateControls.stop()
+    }
+  }, [state])
+
+  return (
+    <>
+      <motion.svg height={2 * Dimension} width={2 * Dimension}>
+        <motion.g
+          animate={rotateControls}
+          transition={{ repeat: Infinity, ease: 'linear' }}
+        >
+          <motion.circle
```

Comment:
I think it's great that we are using this project to play around with things we want to learn, like animations.

To play devil's advocate: a genuine question — do you think this will make the site super slow or look weird on some browsers or mobile?

I think I've been reading too many things like: https://motherfuckingwebsite.com/ (I've lost my sense of beauty).

Reply:
It's a good question, and probably not. We'll run a lighthouse test once we launch, and we can evaluate if the bundle is too big. I highly doubt it though, from a single library